test: Initial work for workspace-type integration tests

ealemayehu opened the pull request:
1. Implemented workspace-types resources
2. Implement integration test for get API

Commit:
Initial work for workspace-type integration tests

1. Implemented workspace-types resources
2. Implement integration test for get API

Commit:
Updated example.yml file

This is for new service-catalog settings

hatimkhan commented on main/integration-tests/support/resources/workspace-types/workspace-types.js:
```diff
@@ -0,0 +1,74 @@
+      name: id,
+      desc: this.setup.gen.description(),
+      status: 'not-approved',
+      product: { productId: this.setup.settings.get('defaultProductId') },
```
instead of this.setup.settings,  can you use this.settings  (the CollectionResource exposes this.settings for you).

Commit:
Made fixes for code review feedback

hatimkhan commented on main/integration-tests/__test__/api-tests/workspace-types/get-workspace-type.test.js:
```diff
@@ -0,0 +1,96 @@
+      });
+    });
+
+    it('should pass to get approved types if user is not an admin', async () => {
```
In most of our tests description, I don't think we say "should pass", maybe we can do something like "should return approved types even if user is not an admin"

hatimkhan commented on main/integration-tests/__test__/api-tests/workspace-types/get-workspace-type.test.js:
```diff
@@ -0,0 +1,96 @@
+      ).resolves.toHaveProperty('id', workspaceTypeId);
+    });
+
+    it('should pass to get no-approved types if user is an admin', async () => {
```
Similar to my comments above, I think the description is a bit hard to read.

ealemayehu replied:
Done

hatimkhan commented on main/integration-tests/__test__/api-tests/workspace-types/get-workspace-type.test.js:
```diff
@@ -0,0 +1,96 @@
+/*
```
I suggest that we add tests for anonymous sessions. There are a few places were we did that, such as the user and study tests.

hatimkhan commented on main/integration-tests/support/resources/workspace-types/workspace-type.js:
```diff
@@ -0,0 +1,35 @@
+  constructor({ clientSession, id, parent }) {
+    super({
+      clientSession,
+      type: 'workspace-type',
```
I think we have been doing camel case for the type, per the second tab in the spreadsheet.

ealemayehu replied:
Done

hatimkhan commented on main/integration-tests/support/resources/workspace-types/workspace-types.js:
```diff
@@ -0,0 +1,73 @@
+  constructor({ clientSession }) {
+    super({
+      clientSession,
+      type: 'workspace-types',
+      childType: 'workspace-type',
```
I think we have been doing camel case for the types per the second tab in the spreadsheet

ealemayehu replied:
Done

Commit:
Added anonymous user test cases

This is to the get api test of workspace-types.
Also made documentation updates.

Commit:
Updated resource names to camel case

Commit:
Fixed typo

SanketD92 reviewed (dismissed):
The initial setup required LGTM. 
(Need to add tests, which I assume will come after this)

hatimkhan committed:
Merge branch 'develop' into ealemayehu/service-workbench-on-aws

hatimkhan commented on main/integration-tests/support/resources/workspace-types/workspace-type.js:
```diff
@@ -0,0 +1,35 @@
+  constructor({ clientSession, id, parent }) {
+    super({
+      clientSession,
+      type: 'WorkspaceType',
```
Sorry, camel case, not pascal case.

ealemayehu replied:
Done

Commit:
Updated resource names to pascal case

Commit:
Merge branch 'ealemayehu/service-workbench-on-aws' of https://github.com/ealemayehu/service-workbench-on-aws into ealemayehu/service-workbench-on-aws

Commit:
Updated resource name to pascal case
